src/components/ImportRecord/ImportRecord.js

```diff
@@ -0,0 +1,102 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { FormattedMessage, injectIntl } from 'react-intl';
+import ky from 'ky';
+import { withOkapiKy, CalloutContext } from '@folio/stripes/core';
+import { LoadingView } from '@folio/stripes/components';
+import withLocation from '../../withLocation';
+
+
+// This is hardwired to the UUID in mod-copycat reference data
+const PROFILE_ID = 'f26df83c-aa25-40b6-876e-96852c3d4fd4';
+
+
+class ImportRecord extends React.Component {
+  static propTypes = {
+    id: PropTypes.string,
+    stripes: PropTypes.shape({
+      logger: PropTypes.object.isRequired,
+    }).isRequired,
+    okapiKy: PropTypes.func.isRequired,
+    intl: PropTypes.shape({
+      formatMessage: PropTypes.func.isRequired,
+    }).isRequired,
+    getParams: PropTypes.func.isRequired,
```

Comment:
Yes! I _hate_ that! We should have a convention that any HOC called `withFoo` should furnish one single prop, `foo` — and that should be an object containing everything that the HOC provides. So for example instead of giving us `props.location` and `props.match` (and others), `withRouter` should just give us `props.router`, and that should contain `location`, `match`, etc.

(What makes the case of `location` even worse is that although it's provided by `getRouter`, we also have a `withLocation` HOC which does _not_ provide `location`!)

Reply:
(And while I'm at it, HOC is a dumb name for this construct.)

Reply:
I agree there are now easier ways to accomplish it specially with hooks.

Reply:
All true critiques of HOCs in general, and why many folks [advocate for render-props over HOCs](https://cdb.reacttraining.com/use-a-render-prop-50de598f11ce). He even calls out `react-router` in this example, though for different reasons.